pkg/shortnames/shortnames.go

```diff
@@ -246,6 +250,21 @@ func Resolve(ctx *types.SystemContext, name string) (*Resolved, error) {
 	}
 	resolved.systemContext = ctx
 
+	// Resolve to docker.io only if enforced by the caller (e.g., Podman's
+	// Docker-compatible REST API).
+	if ctx.ResolveShortNamesToDockerHub {
+		named, err := reference.ParseNormalizedNamed(name)
+		if err != nil {
+			return nil, errors.Wrapf(err, "cannot normalize input: %q", name)
+		}
+		// Make sure to add ":latest" if needed
+		named = reference.TagNameOnly(named)
```

Comment:
Please make the `TagNameOnly` call a responsibility of `addCandidate`, to reduce duplication now that there are going to be 4 copies.